src/field_theory/fixed.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import algebra.group_action_hom
+import field_theory.normal
+import field_theory.subfield
+import ring_theory.polynomial
+
+/-!
+# Fixed field under a group action.
+
+This is the basis of the Fundamental Theorem of Galois Theory.
+Given a (finite) group `G` that acts on a field `F`, we define `fixed_points G F`,
+the subfield consisting of elements of `F` fixed_points by every element of `G`.
+
+This subfield is then normal (proved in this file) and separable (TODO), and in addition
+if `G` acts faithfully on `F` then `findim (fixed_points G F) F = fintype.card G` (TODO).
+
+## Main Deifnitions
+
+- `fixed_points G F`, the subfield consisting of elements of `F` fixed_points by every element of `G`, where
+`G` is a group that acts on `F`.
+
+-/
+
+noncomputable theory
+open_locale classical
+open mul_action
+
+universes u v w
+
+variables (G : Type u) [group G] (F : Type v) [field F] [mul_semiring_action G F] (g : G)
+
+/-- The subfield fixed_points by one element of the group. -/
+def fixed_by : set F :=
```

Comment:
Should this definition be given in greater generality somewhere close to the definition of `fixed_points`? I guess we might want to use it in places where `F` is not a field (but a representation, say).

Reply:
done